pandas/core/resample.py

```diff
@@ -2671,6 +2685,8 @@ def asfreq(
         if how is None:
             how = "E"
 
+        freq = freq_to_period_freqstr(obj.index.freq.n, freq)
```

Comment:
not sure about this one:

```python
In [1]: ser = pd.Series([1, 2], index=pd.PeriodIndex(['2020-01', '2020-03'], freq='2M'))
   ...: print(ser.asfreq('M'))
2020-02    1
2020-04    2
Freq: 2M, dtype: int64
```

I think:
- if `freq` is an offset, use `freq.n` and `freq.name`
- if `freq` is a string, then it require that it be a period alias already

Any, it's a bit concerning that no test fails - indicating that there wasn't good test coverage here to begin with

Could we make a separate PR which adds a test with the above snippet please?